rcp/org.testeditor.rcp4/src/org/testeditor/rcp4/dialogs/NetworkConnectionSettingDialog.java

```diff
@@ -0,0 +1,316 @@
+		noProxyHosts = new Text(container, SWT.BORDER);
+		noProxyHosts.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+		noProxyHosts.setText(noProxyHostsSetting);
+		noProxyHosts.addModifyListener(new ModifyListener() {
+
+			@Override
+			public void modifyText(ModifyEvent e) {
+				noProxyHostsSetting = noProxyHosts.getText();
+				updateNetworkState();
+			}
+		});
+	}
+
+	private void createMavenSettingsWidgets(final Composite container) {
+		new Label(container, SWT.NORMAL).setText("Path to maven settings:");
+		Composite mavenPathControl = new Composite(container, SWT.NORMAL);
+		mavenPathControl.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+		mavenPathControl.setLayout(new GridLayout(2, false));
+		pathToMavenSettingsFile = new Text(mavenPathControl, SWT.BORDER);
+		pathToMavenSettingsFile.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+		pathToMavenSettingsFile.setText(pathToMavenSettingsFileSettings);
+		pathToMavenSettingsFile.addModifyListener(new ModifyListener() {
+
+			@Override
+			public void modifyText(ModifyEvent e) {
+				pathToMavenSettingsFileSettings = pathToMavenSettingsFile.getText();
+			}
+		});
+		Button fileSelection = new Button(mavenPathControl, SWT.BORDER);
+		fileSelection.setText("...");
+		fileSelection.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				FileDialog fileDialog = new FileDialog(container.getShell());
+				fileDialog.setText("Select Maven settings");
+				String path = fileDialog.open();
+				if (path != null) {
+					pathToMavenSettingsFile.setText(path);
+				}
+			}
+		});
+	}
+
+	private void updateNetworkState() {
+		Runnable runnable = new Runnable() {
+
+			@Override
+			public void run() {
+				if (isInternetAvailable(true)) {
+					if (workOffline) {
+						updateNetworkSateWidget("work offline", SWT.COLOR_DARK_YELLOW);
+					} else {
+						updateNetworkSateWidget("connected", SWT.COLOR_DARK_GREEN);
+					}
+				} else {
+					updateNetworkSateWidget("diconnected", SWT.COLOR_RED);
+				}
+			}
+		};
+		new Thread(runnable).start();
+	}
+
+	protected void updateNetworkSateWidget(final String message, final int color) {
+		Display.getDefault().asyncExec(new Runnable() {
+
+			@Override
+			public void run() {
+				if (!internetState.isDisposed()) {
+					internetState.setText(message);
+					internetState.setForeground(Display.getDefault().getSystemColor(color));
+					internetState.getParent().layout(true);
+				}
+			}
+		});
+	}
+
+	@Override
+	protected void configureShell(Shell newShell) {
+		super.configureShell(newShell);
+		newShell.setText("Network configuration dialog");
+	}
+
+	@Override
+	protected Point getInitialSize() {
+		return new Point(500, 380);
+	}
+
+	@Override
+	protected void okPressed() {
+		prefs.putBoolean(WORKOFFLINE, workOffline);
+		prefs.put(PATH_TO_MAVENSETTINGS, pathToMavenSettingsFileSettings);
+		prefs.put(PROXY_HOST, proxyHostSetting);
```

Comment:
Ok I added the same config values to the https variables. To keep the ui easy, i prefer to have one proxy config until somebody needs a seond one.